packages/lit-task-auto-top-up/src/taskHelpers.ts

```diff
@@ -0,0 +1,46 @@
+import { Job } from '@hokify/agenda';
+import consola from 'consola';
+import _ from 'lodash';
+import VError, { MultiError } from 'verror';
+
+import { TaskResult } from './types/types';
+
+export function printTaskResultsAndFailures(results: PromiseSettledResult<TaskResult>[]) {
+  const fulfilled = results.filter(
+    (r): r is PromiseFulfilledResult<TaskResult> => r.status === 'fulfilled'
+  );
+  const rejected = results.filter((r): r is PromiseRejectedResult => r.status === 'rejected');
+
+  const errors = rejected.map(({ reason }: { reason: Error }) => reason);
+  const successes = fulfilled.map(({ value }) => value);
+
+  consola.log(`Succeeded topping off ${successes.length} recipients`);
+  _.forEach(successes, (r) => {
+    const { recipientAddress, ...rest } = r;
+    consola.log(`Minted for ${recipientAddress}`, rest);
+  });
+
+  if (errors.length > 0) {
+    consola.log(`Failed to top off ${errors.length} recipients`);
+    _.forEach(errors, (error) => {
+      consola.error(error, JSON.stringify(VError.info(error), null, 2));
+    });
+
+    if (successes.length > 1 && successes.length === errors.length) {
```

Comment:
```suggestion
    if (results.length === errors.length) {
```
I guess that was the intent based on the comment

Reply:
🤦 you're right :) Fixed.